holidays/countries/united_kingdom.py

```diff
@@ -51,124 +46,157 @@ class UnitedKingdom(HolidayBase):
         "Wales",
     )
 
-    def __init__(self, **kwargs: Any) -> None:
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
         # default subdiv to UK; state for backwards compatibility
         if not kwargs.get("subdiv", kwargs.get("state")):
             kwargs["subdiv"] = "UK"
-        HolidayBase.__init__(self, **kwargs)
+        super().__init__(*args, **kwargs)
+
+    def _add_observed(self, dt: date) -> None:
+        if self.observed and self._is_weekend(dt):
+            obs_date = _get_nth_weekday_from(1, MON, dt)
+            if obs_date in self:
+                obs_date += td(days=+1)
+            self._add_holiday("%s (Observed)" % self[dt], obs_date)
 
     def _populate(self, year: int) -> None:
         super()._populate(year)
 
+        # Good Friday
+        self._add_good_friday("Good Friday")
+
+        # May Day bank holiday (first Monday in May)
+        if year >= 1978:
+            dt = (
+                date(year, MAY, 8)
+                # In 2020 moved to Friday to mark 75th anniversary of VE Day.
+                if year in {1995, 2020}
+                else _get_nth_weekday_of_month(1, MON, MAY, year)
+            )
+            self._add_holiday("May Day", dt)
+
+        # Spring bank holiday (last Monday in May)
+        if year >= 1971:
+            spring_bank_dates = {
+                2002: date(year, JUN, 4),
+                2012: date(year, JUN, 4),
+                2022: date(year, JUN, 2),
+            }
+            dt = spring_bank_dates.get(
+                year, _get_nth_weekday_of_month(-1, MON, MAY, year)
+            )
+            self._add_holiday("Spring Bank Holiday", dt)
+
+        # Christmas Day
+        dec_25 = self._add_christmas_day("Christmas Day")
+
+        # Boxing Day
+        dec_26 = self._add_christmas_day_two("Boxing Day")
+
+        self._add_observed(dec_25)
+        self._add_observed(dec_26)
+
+    def _add_subdiv_holidays(self):
         # New Year's Day
-        if year >= 1974:
-            name = "New Year's Day"
-            dt = date(year, JAN, 1)
-            self[dt] = name
-            if self.observed and self._is_weekend(dt):
-                self[_get_nth_weekday_from(1, MON, dt)] = name + " (Observed)"
+        if self._year >= 1974:
+            self._add_observed(self._add_new_years_day("New Year's Day"))
```

Comment:
Can it be a part of `_populate`?

Reply:
For proper calculation of observed "New Year Holiday" in Scotland (Jan 2), "New Year's Day" should already exist before Scotland holidays population. If it were possible to populate "subdiv" holidays **after** common, it would be simpler here (and in Australia with Anzac Day and Christmas).

Reply:
Gotcha! It sounds like something worth remembering while working on https://github.com/dr-prodigy/python-holidays/issues/1252 implementation

No way to move it atm, so keeping as is.